<!-- move base -->
<launch>
  <master auto="start"/>
    
  <arg name="slam_method" default="gmapping"/>   
  <!--arg name="slam_method" value="$(arg slam_method)"/-->
  <include file="$(find autoexplorer)/launch/includes/start_$(arg slam_method).launch">
  </include>
  
  <!--include file="$(find autoexplorer)/launch/includes/start_gmapping.launch">
  </include-->

  
  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <!--remap from="odom" to="base_footprint"/-->
    <!--param name="footprint_padding" value="0.01" /-->
    <param name="controller_frequency" value="5.0" />
    <!--param name="controller_patience" value="1.0" /--> <!--3.0-->
    <param name="oscillation_timeout" value="30.0" />
    <param name="oscillation_distance" value="0.5" />
    <param name="planner_patience" value="1" />
    <param name="controller_patience" value="1" /> 
    <!--remap from="cmd_vel" to="mobile_base/commands/velocity"/-->
    <!--remap from="cmd_vel" to="move_base_node/cmd_vel"/-->
        
    <!--param name="recovery_behavior_enabled" value="false" /-->
    <rosparam file="$(find autoexplorer)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find autoexplorer)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find autoexplorer)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find autoexplorer)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find autoexplorer)/param/base_local_planner_params.yaml" command="load" />  
    <param name="global_costmap/global_frame" value="map"/>
    <param name="global_costmap/robot_base_frame" value="base_link"/> 
    <param name="local_costmap/global_frame" value="map"/>
    <param name="local_costmap/robot_base_frame" value="base_link"/>
  </node>

</launch>
